package-build/tools/vcpkgInstall.ps1

```diff
@@ -5,33 +5,66 @@ $RootDir = 'C:\opt'
 $InstallDir = "$RootDir\vcpkg"
 $VcpkgVersion = "2020.04"
 
-try {
+try
+{
   New-Item -ItemType Directory -Path $RootDir -Force | Out-Null
 
+  # shadow the Vcpkg environment variables which could break the deployment.
+  if (Test-Path env:VCPKG_ROOT)
+  {
+    Write-Warning "%VCPKG_ROOT% is defined. Removing from current session."
+    Remove-Item env:VCPKG_ROOT
+  }
+
   Write-Host 'Deploying vcpkg...'
-  if (-Not (Test-Path -Path "$InstallDir\vcpkg.exe" -PathType Leaf)) {
-    if (-Not (Test-Path -Path $InstallDir -PathType Container)) {
+  if (-Not (Test-Path -Path "$InstallDir\vcpkg.exe" -PathType Leaf))
+  {
+    if (-Not (Test-Path -Path $InstallDir -PathType Container))
+    {
       $Env:GIT_REDIRECT_STDERR="2>&1"
-      git clone --depth=1 $Uri $InstallDir -q -b $VcpkgVersion | Out-null
-      # Chocolatey bloats %PATH% by 3x.
-      # It increases the odd running over the environment value size limit.
-      # So we skip the bootstrap-vcpkg.bat and invoke the powershell one to
-      # avoid that.
-      Invoke-Expression "$InstallDir\scripts\bootstrap.ps1"
-    } else {
+      git clone $Uri $InstallDir -q -b $VcpkgVersion | Out-Null
+      & "$InstallDir\bootstrap-vcpkg.bat"
+    }
+    else
+    {
       throw "Found incomplete Vcpkg installation. Remove $InstallDir and reinstall again."
     }
-  } else {
-    Write-Host 'Vcpkg.exe is found. Skip deploying vcpkg...'
+  }
+  else
+  {
+    Write-Host "Vcpkg.exe is found. Upgrading it to '$VcpkgVersion'."
+    Write-Host "Checking any uncommitted changes under '$InstallDir'."
+
+    $gitStatus = & git "--git-dir=$InstallDir\.git" "--work-tree=$InstallDir" status --porcelain
+    if (-not ([string]::IsNullOrEmpty($gitStatus)))
+    {
+      Write-Warning $gitStatus
+      Write-Warning "Uncommitted changes are found."
```

Comment:
vcpkg cannot be upgraded because there are changes in the repository. Please commit them or revert them, then restart the ROS installation. Because this is required for building ROS packages, ROS install cannot continue.